subsys/bluetooth/controller/hci_cmd_put.c

```diff
@@ -0,0 +1,851 @@
+	cmds->hci_le_set_scan_enable = 1;
+#endif
+
+#if defined(CONFIG_BT_CENTRAL)
+	cmds->hci_le_create_connection = 1;
+	cmds->hci_le_create_connection_cancel = 1;
+#endif
+
+	cmds->hci_le_read_white_list_size = 1;
+	cmds->hci_le_clear_white_list = 1;
+	cmds->hci_le_add_device_to_white_list = 1;
+	cmds->hci_le_remove_device_from_white_list = 1;
+
+#if defined(CONFIG_BT_CENTRAL)
+	cmds->hci_le_connection_update = 1;
+#endif
+
+#if defined(CONFIG_BT_CENTRAL) || defined(CONFIG_BT_CTLR_ADV_EXT)
+	cmds->hci_le_set_host_channel_classification = 1;
+#endif
+
+#if defined(CONFIG_BT_CONN)
+	cmds->hci_le_read_channel_map = 1;
+	cmds->hci_le_read_remote_features = 1;
+#endif
+
+	cmds->hci_le_encrypt = 1;
+	cmds->hci_le_rand = 1;
+
+#if defined(CONFIG_BT_CTLR_LE_ENC)
+	cmds->hci_le_enable_encryption = 1;
+	cmds->hci_le_long_term_key_request_reply = 1;
+	cmds->hci_le_long_term_key_request_negative_reply = 1;
+#endif
+
+	cmds->hci_le_read_supported_states = 1;
+
+#if defined(BT_CTLR_DTM)
+	cmds->hci_le_receiver_test_v1 = 1;
+	cmds->hci_le_transmitter_test_v1 = 1;
+	cmds->hci_le_test_end = 1;
+#endif
+
+#if defined(CONFIG_BT_CTLR_DATA_LENGTH)
+	cmds->hci_le_set_data_length = 1;
+	cmds->hci_le_read_suggested_default_data_length = 1;
+	cmds->hci_le_write_suggested_default_data_length = 1;
+#endif
+
+#if defined(CONFIG_BT_CTLR_PRIVACY)
+	cmds->hci_le_add_device_to_resolving_list = 1;
+	cmds->hci_le_remove_device_from_resolving_list = 1;
+	cmds->hci_le_clear_resolving_list = 1;
+	cmds->hci_le_read_resolving_list_size = 1;
+	cmds->hci_le_set_address_resolution_enable = 1;
+	cmds->hci_le_set_resolvable_private_address_timeout = 1;
+#endif
+
+#if defined(CONFIG_BT_CTLR_DATA_LENGTH)
+	cmds->hci_le_read_maximum_data_length = 1;
+#endif
+
+#if defined(CONFIG_BT_CTLR_PHY)
+	cmds->hci_le_read_phy = 1;
+	cmds->hci_le_set_default_phy = 1;
+	cmds->hci_le_set_phy = 1;
+#endif
+
+#if defined(BT_CTLR_DTM)
+	cmds->hci_le_receiver_test_v2 = 1;
+	cmds->hci_le_transmitter_test_v2 = 1;
+#endif
+
+#if defined(CONFIG_BT_CTLR_ADV_EXT)
+#if defined(CONFIG_BT_BROADCASTER)
+	cmds->hci_le_set_advertising_set_random_address = 1;
+	cmds->hci_le_set_extended_advertising_parameters = 1;
+	cmds->hci_le_set_extended_advertising_data = 1;
+	cmds->hci_le_set_extended_scan_response_data = 1;
+	cmds->hci_le_set_extended_advertising_enable = 1;
+	cmds->hci_le_read_maximum_advertising_data_length = 1;
+	cmds->hci_le_read_number_of_supported_advertising_sets = 1;
+	cmds->hci_le_remove_advertising_set = 1;
+	cmds->hci_le_clear_advertising_sets = 1;
+#endif
+
+#if defined(CONFIG_BT_OBSERVER)
+	cmds->hci_le_set_extended_scan_parameters = 1;
+	cmds->hci_le_set_extended_scan_enable = 1;
+#endif
+#if defined(CONFIG_BT_CENTRAL)
+	cmds->hci_le_extended_create_connection = 1;
+#endif
+#endif
+	cmds->hci_le_read_transmit_power = 1;
+
+#if defined(CONFIG_BT_CTLR_PRIVACY)
+	cmds->hci_le_set_privacy_mode = 1;
+#endif
+}
+
+static void le_read_supported_states(uint64_t *buf)
+{
+	uint64_t states = 0U;
```

Comment:
Do we use uint64_t somewhere else? It usually add quite some code.

Reply:
I agree that we should try to avoid the use of uint64_t. However, in this case it really makes sense to use it

Reply:
Could you please compare the sizes of this implemnetation vs implementation w/o uint64_t?

Reply:
Using 2*uint32 saves 100 bytes of code. I will therefore update this function.

- [ ] Use 2*uint32_t